packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -959,3 +923,1325 @@ class _RenderTheatre extends RenderBox with ContainerRenderObjectMixin<RenderBox
+  _RenderTheatre get _theatre => _theatreChildModel._theatre;
+
+  @override
+  void _addToChildModel(_RenderDeferredLayoutBox child) {
+    assert(_overlayChildRenderBox == null, 'Failed to add $child. $_overlayChildRenderBox is attached to $this');
+    assert(_showTimestamp != null);
+    _overlayChildRenderBox = child;
+    _theatreChildModel._add(this);
+  }
+
+  @override
+  void _removeFromChildModel(_RenderDeferredLayoutBox child) {
+    assert(child == _overlayChildRenderBox);
+    _overlayChildRenderBox = null;
+    assert(_theatreChildModel._sortedChildren.contains(this));
+    _theatreChildModel._remove(this);
+  }
+
+  @override
+  void _activate(_RenderDeferredLayoutBox child) {
+    assert(_debugNotDisposed());
+    assert(_overlayChildRenderBox == null, '$_overlayChildRenderBox');
+    super._activate(child);
+    _overlayChildRenderBox = child;
+  }
+
+  @override
+  void _deactivate(_RenderDeferredLayoutBox child) {
+    assert(_debugNotDisposed());
+    super._deactivate(child);
+    _overlayChildRenderBox = null;
+  }
+
+  @override
+  String toString() => '$_showTimestamp';
+}
+
+class _RenderTheatreMarker extends InheritedWidget {
+  const _RenderTheatreMarker({
+    required this.theatre,
+    required this.overlayEntryWidgetState,
+    required super.child,
+  });
+
+  final _RenderTheatre theatre;
+  final _OverlayEntryWidgetState overlayEntryWidgetState;
+
+  @override
+  bool updateShouldNotify(_RenderTheatreMarker oldWidget) {
+    return oldWidget.theatre != theatre
+        || oldWidget.overlayEntryWidgetState != overlayEntryWidgetState;
+  }
+
+  @override
+  InheritedElement createElement() => _RenderTheatreMarkerElement(this);
+
+  static _RenderTheatreMarker? of(BuildContext context, { required bool targetRootOverlay }) {
+    final _RenderTheatreMarkerElement? element = context.getElementForInheritedWidgetOfExactType<_RenderTheatreMarker>() as _RenderTheatreMarkerElement?;
+    if (element == null) {
+      return null;
+    }
+    final _RenderTheatreMarkerElement dependency = targetRootOverlay ? element._topRenderTheatre : element;
+    return context.dependOnInheritedElement(dependency) as _RenderTheatreMarker;
+  }
+}
+
+class _RenderTheatreMarkerElement extends InheritedElement {
+  _RenderTheatreMarkerElement(super.widget);
+
+  _RenderTheatreMarkerElement get _topRenderTheatre {
+    _RenderTheatreMarkerElement? ancestor;
+    visitAncestorElements((Element element) {
+      assert(ancestor == null);
+      ancestor = element.getElementForInheritedWidgetOfExactType<_RenderTheatreMarker>() as _RenderTheatreMarkerElement?;
+      return false;
+    });
+    return ancestor?._topRenderTheatre ?? this;
+  }
+}
+
+class _OverlayPortalState extends State<OverlayPortal> {
+  // The developer must call `show` to reveal the overlay so we can get the
+  // timestamp of the user interaction for sorting.
+  _TopmostLocation? _location;
+  int? _showTimestamp;
+  // See https://github.com/dart-lang/sdk/issues/41717.
```

Comment:
unless this is a TODO, it's better to just inline the comment. The code will probably survive the death of GitHub.

Reply:
(if it's a TODO, we should label it as such)

Reply:
removed

Reply:
But I have other code comments that contain links to things hosted on github.